homeassistant/helpers/entity_platform.py

```diff
@@ -461,10 +472,10 @@ async def _async_add_entity(  # type: ignore[no-untyped-def]  # noqa: C901
                     "suggested_area",
                 ):
                     if key in device_info:
-                        processed_dev_info[key] = device_info[key]
+                        processed_dev_info[key] = device_info[key]  # type: ignore[misc]
 
                 try:
-                    device = device_registry.async_get_or_create(**processed_dev_info)
+                    device = device_registry.async_get_or_create(**processed_dev_info)  # type: ignore[arg-type]
```

Comment:
@scop I think this should be rewritten using new `DeviceInfo` typed dict. And `device_registry.async_get_or_create` should just take it as input to avoid listing the same keys many times.